### DISCUSSION TITLE
Custom devices

### DISCUSSION THREAD

**Discussion by tidharmor** [make-all/tuya-local#1190] [General]
Hi @make-all ,

I previously opened an issue about a [Water Leak Detector](https://github.com/make-all/tuya-local/issues/1184) but you closed it because you said the integration doesn't support this kind of device.

First, I would like you to reconsider.
I managed to add the device and got an alarm when I triggered it using this integration.
Is it possible to add it anyway so I don't lose it every time the integration is updated?

Second, is it possible to add a custom devices folder directly under /config, similar to ZHA custom quirks folder?
This way if I have a device which is not supported, but it's working for me, I can keep using it without it being deleted every time the integration is updated.

Thanks.

**Comment by make-all** [comment]
Battery powered wifi devices are a support nightmare.  I won't be changing my decision not to support them, as there is no way to do so reliably, and to add support that is unreliable will cost too much in time supporting users through the process of connecting and dealing with the complaints about them not reposting reliably.
